### ISSUE TITLE
False Negative: FinalLocalVariableCheck does not report for variables assigned in adjacent catches

### ISSUE THREAD

**Issue by lithops-zty** [checkstyle/checkstyle#19717]
I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java 
/var/tmp $ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalLocalVariableCheck"/>
    </module>
</module>
/var/tmp $ cat Test.java 
import java.io.FileNotFoundException;
import java.nio.channels.FileLock;

public class Test {
    public void test(boolean x) {
        int u;  // violation, but not reported
        try {
        } catch (IndexOutOfBoundsException e) {
            u = 9;
        } catch (RuntimeException e) {
            u = 10;
        }

        int v;  // violation
        if (x) {
            v = 0;
        } else {
            v = 1;
        }
    }

}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.4.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /mnt/d/IntellijProjects/checkstyle-playground/inputs/final-local-variable/Test.java:14:13: Variable 'v' should be declared final. [FinalLocalVariable]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

A violation on `u` not declared final should be reported, since catch clauses are mutually-exclusive.

**Comment by lithops-zty** [comment]
I'd like to work on it if it is approved.

**Comment by romani** [comment]
Please fix link to documentation of Check

**Comment by lithops-zty** [comment]
Link fixed. 
I'm on it.

**Comment by romani** [comment]
>I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html

I still see this. Not good link.

**Comment by lithops-zty** [comment]
Sorry, GitHub did not save changes successfully. Now it is fixed.

**Comment by romani** [comment]
We decided to not fix this as it is grey area of compilation.
Better to avoid false positives than validation might cause.

See discussion at https://github.com/checkstyle/checkstyle/pull/19728 on why we decided to not do this
